src/Compilers/Core/Portable/RealParser.cs

```diff
@@ -358,14 +342,24 @@ public static DecimalFloatingPointString FromSource(string source)
                     int firstExponent = i;
                     int lastExponent = i;
                     while (i < source.Length && source[i] >= '0' && source[i] <= '9') lastExponent = ++i;
-                    int exponentMagnitude = int.Parse(source.Substring(firstExponent, lastExponent - firstExponent));
-                    if (exponentSign == '-')
+
+                    int exponentMagnitude = 0;
+
+                    if (int.TryParse(source.Substring(firstExponent, lastExponent - firstExponent), out exponentMagnitude) &&
+                        exponentMagnitude <= MAX_EXP_QUAD)
                     {
-                        exponent -= exponentMagnitude;
+                        if (exponentSign == '-')
+                        {
+                            exponent -= exponentMagnitude;
+                        }
+                        else
+                        {
+                            exponent += exponentMagnitude;
```

Comment:
@jaredpar + @gafter: You would need a 2147467264 digit number. 

Sure that would make many users happy :D 

I chose the cutoff for the max exponent for a quadrature floating point number just in case. 10K or something would have been fine too. Not going to be a problem for at least 10 years.